whylog/front/__init__.py

```diff
@@ -0,0 +1,37 @@
+from whylog.client import WhylogClient
+from whylog.teacher import WhylogTeacher, WhylogChecker
+
+
+class WhylogFront(object):
+    """
+    This class is the only one that will be used by plugins for text editors
+    implementing Whylog.
+    It provide all funtionalities of whylog.
+    """
+
+    def __init__(self, **opening_params):
+        self._whylog_client = WhylogClient(**opening_params)
+        self._whylog_teacher = WhylogTeacher(**opening_params)
+
+    def get_causes(self, front_input):
+        """
+        Function returning result of the investigation of the front input
+        """
+        return self._whylog_client.get_causes(front_input)
+
+    def get_tree(self, front_input):
+        return self._whylog_client.get_tree(front_input)
+
+    def new_teach_object(self):
+        """
+        This function return object that can teach new dependence for whylog
+        """
+        return self._whylog_teacher.new_teach_object()
+
+    @staticmethod
+    def check_teach_object(teach_object, front_input):
+        check_client = WhylogChecker(teach_object)
+        return check_client.get_causes(front_input)
```

Comment:
I think this should use WhylogClient with passing the teach object into it and then the teach object needs to be passed deep down into the config layer so that it is taken into account as a temporary set of rules and parsers. This will complicate the code somewhat and I think we could maybe skip the requirement of testing the new teach object in the first version - program will still be usable and the implementation will be much easier to understand.